pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1272,8 +1272,17 @@ self: super: {
   # upstream issue: https://github.com/vmchale/atspkg/issues/12
   language-ats = dontCheck super.language-ats;
 
-  # polysemy has occasional test failures from what looks like buggy async tests.
-  # We think this will probably be fixed when updating to the polysemy version in LTS-15.
-  polysemy = dontCheck super.polysemy;
+  # polysemy-plugin requires polysemy >= 1.2.0.0
+  polysemy = self.polysemy_1_2_2_0;
```

Comment:
The problem with bumping `polysemy` to 1.2.2.0 is that it is from the LTS-14 package set, so bumping may cause any other packages in LTS-14 depending on it to fail to build:

https://github.com/NixOS/nixpkgs/blob/e8a36cdf57193e56514aa6eeff936159372f0ace/pkgs/development/haskell-modules/configuration-hackage2nix.yaml#L1609

Instead, what you should do is edit polysemy-plugin to depend on `polysemy_1_2_2_0` instead of `polysemy`:

```diff
diff --git a/pkgs/development/haskell-modules/configuration-common.nix b/pkgs/development/haskell-modules/configuration-common.nix
index 8bea3fc73d7..566aaa9c025 100644
--- a/pkgs/development/haskell-modules/configuration-common.nix
+++ b/pkgs/development/haskell-modules/configuration-common.nix
@@ -1276,4 +1276,9 @@ self: super: {
   # We think this will probably be fixed when updating to the polysemy version in LTS-15.
   polysemy = dontCheck super.polysemy;
 
+  # polysemy is locked to version 1.0.0.0 from LTS-14, but polysemy-plugin needs a
+  # later version of polysemy.  This override can probably be dropped when we move
+  # to LTS-15.
+  polysemy-plugin = super.polysemy-plugin.override { polysemy = self.polysemy_1_2_2_0; };
+
 } // import ./configuration-tensorflow.nix {inherit pkgs haskellLib;} self super
```

Reply:
It is true that we normally don't deviate from the LTS version recommendations. It is not unheard of, though, that we do. :-) For some packages, we just want the latest versions even though it might cause some trouble down the road. The `update-stackage` script even has a [hard-coded list of packages where we just ignore Stackage](https://github.com/NixOS/cabal2nix/blob/master/update-stackage.sh#L26-L43). In case of `polysemy`, I have mixed feelings about deviating from Stackage. On one hand, the package does have very few dependencies. According to http://packdeps.haskellers.com/reverse/polysemy:

![screenshot](https://user-images.githubusercontent.com/28323/67616482-a5045800-f7d9-11e9-8b65-db5d76200c95.jpg)

This means that we can probably deviate from Stackage without breaking anything.

On the other hand, the small number of dependencies also suggests that `polysemy` is probably not that important, really, and deviating from Stackage just to support the latest version might be overkill since it's not clear whether many people are using it.

Reply:
Ok, so some additional context:

Of the packages that depend on polysemy, `gamgee` is the only package that would be truly affected by a bump to 1.1:
  - axel: currently broken, fixed with bump to 1.1
  - co-log-polysemy: currently broken, will remain broken.
  - gamgee: marked as broken, though builds successfully currently. Will be broken by bump to 1.1.
  - knit-haskell: currently broken, fixed with these changes.
  - polysemy-RandomFu: currently broken, fixed with these changes.
  - polysemy-zoo: currently broken, fixed with these changes.

So that would lean me towards pushing polysemy up to 1.1. I could probably submit a PR to make gamgee work with 1.1.

At the same time, I agree with the statement that "polysemy isn't very important/not many people are using it". Despite being 1.*, it's still a fairly new package. I'm using it primarily out of interest/research.

Reply:
With this new information, what's your opinion on this @peti?

Reply:
I don't have any strong opinions on this matter. Whatever solution works for you is fine, IMHO.